### PR TITLE
(RE-6743) Add Puppet-Agent registry entries

### PR DESCRIPTION
**Pull request by jcoconnor**
These will over-ride the generic set of registry entries in Vanagon

### CODE REVIEW

**Comment by McdonaldSeanp** [comment]
:+1: seems simple enough

File: resources/windows/wix/registryEntries.wxs.erb [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      </Component>
+
+      <Component
+        Id="RegistryEntriesArchitectureDependent"
+        Directory="TARGETDIR"
+        Guid="E6D5AF4F-ACC4-4D11-AFCE-299A9ED2152C"
+        Win64="<%= settings[:win64] %>"
+        Permanent="yes">
+
+        <RegistryKey
```
Comment (McdonaldSeanp) [inline resources/windows/wix/registryEntries.wxs.erb]: The following seems verbatim from https://github.com/jcoconnor/vanagon/blob/ticket/master/RE-6743-Registry/resources/windows/wix/registryEntries.wxs.erb#L18L43 , is that on purpose?

Comment (McdonaldSeanp) [inline resources/windows/wix/registryEntries.wxs.erb]: It's actually not just this, it's the entire component from line 62 on

Comment (McdonaldSeanp) [inline resources/windows/wix/registryEntries.wxs.erb]: Wait no nevermind, this is supposed to override the vanagon file

**Comment by McdonaldSeanp** [comment]
:+1:

File: resources/windows/wix/registryEntries.wxs.erb (hunk 1 above)
Comment (jcoconnor) [inline resources/windows/wix/registryEntries.wxs.erb]: Don't follow you @McdonaldSeanp about line 62, but it is deliberate - remember - this file will override whats in Vanagon, so anything in there (as an example) will have to be repeated here.

Hold off on merge of this - its not quite compiling yet, so have put a Do Not Merge on it.

Comment (jcoconnor) [inline resources/windows/wix/registryEntries.wxs.erb]: You got it!!

File: resources/windows/wix/registryEntries.wxs.erb [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    Puppet Agent Specific Registry Entries
+    This overrides the generic file in the vanagon directory
+  -->
+  <Fragment>
+    <ComponentGroup Id="RegistryComponentGroup">
+      <Component Id="RegistryEntries" Guid="68F5D91C-49C8-43F7-940A-481007689E79" Win64="no">
+        <RegistryKey
+          Root="HKLM"
+          Key="SOFTWARE\<%= settings[:company_name] %>\<%= settings[:product_name] %>"
+          Action="create" >
```
Comment (jcoconnor) [inline resources/windows/wix/registryEntries.wxs.erb]: @ferventcoder @Iristyle Question here - the component/registry entry on line 62 has ```Permanent="yes"``` set.
Should we be doing same here ?

**Comment by jcoconnor** [comment]
@McdonaldSeanp @melissa @jpogran  - Have fixed compile error - this should be ready for merge now.

File: resources/windows/wix/registryEntries.wxs.erb (hunk 2 above)
Comment (jcoconnor) [inline resources/windows/wix/registryEntries.wxs.erb]: Got hipchat response back from @ferventcoder  - we should leave this as is.

**Pull request merged**